samples/gpu/video_reader.cpp

```diff
@@ -20,24 +21,26 @@ int main(int argc, const char* argv[])
         return -1;
 
     const std::string fname(argv[1]);
+    cv::VideoCapture reader(fname);
+    cv::Size imageSize{(int)reader.get(cv::CAP_PROP_FRAME_WIDTH),
+        (int)reader.get(cv::CAP_PROP_FRAME_HEIGHT)};
 
     cv::namedWindow("CPU", cv::WINDOW_NORMAL);
 #if defined(HAVE_OPENGL)
```

Comment:
Should this also check for the presence of QT6 if this crashes the sampel without it?

Reply:
The `reader` nor the `imageSize` is dependent on the QT6. Is it crashing on this?

Reply:
The reader is moved in earlier from below from the previous sample code, to extract the image size of the frames.

Reply:
```
#if defined(HAVE_OPENGL)
        cv::imshow("GPU", d_frame);
#else
        d_frame.download(frame);
        cv::imshow("GPU", frame);
#endif
```

According to your previous comment, if OpenCV is built against OpenGL but not against QT6 `cv::imshow("GPU", d_frame) will cause the sample to crash.  To prevent this you need to be able to check for QT6 at compile or runtime. e.g.
```
#if defined(HAVE_OPENGL) && defined(HAVE_QT6)
```

Reply:
I haven't checked for other GUI backends and I was assuming to work similarly. But I can add the check to enforce QT6 guard.

Reply:
Where is `HAVE_QT6` defined? It is not included in `#include "cvconfig.h"`.

Reply:
Where is `HAVE_QT6` defined? It is not in `#include "cvconfig.h"`.

Reply:
> Where is `HAVE_QT6` defined? It is not in `#include "cvconfig.h"`.

I don't think it is, it was an example of the test that needs to be performed.

The point of my comments is that you have taken a sample which doen't crash when run and converted it into a sample that will crash when using the default GUI backend which most users will have compiled OpenCV against.  Do you think this will be useful to people who run the sample who don't realize they need QT6, or do you think they will see the crash and assume the `cudacodec::VideoReader` class has a bug?

Reply:
Maybe I should split the sample into two then, one purely for demonstrating `cudacodec::VideoReader`  parsing, and other using opengl+gpumat. Would this be clearer?

Reply:
If the new sample still crashes then I don't think its clearer.  

There are several options, not limited to:
1) Find a way to check for working GUI backends at compile/runtime.  If the GUI's are plugins then I would have thought you could do this at runtime.
2) Check for exceptions and print an informative message combined with a flag to enable or disable calling `cv::imshow` with a `GpuMat`.
3) ...

Reply:
I have now introduced a new flag `HAVE_QT_OPENGL` for checking both qt and opengl in the new commit. Please have a look.